gunpowder/nodes/astype.py

```diff
@@ -0,0 +1,62 @@
+from .batch_filter import BatchFilter
+from gunpowder.array import ArrayKey, Array
+from gunpowder.batch import Batch
+import logging
+
+logger = logging.getLogger(__name__)
+
+class AsType(BatchFilter):
```

Comment:
This could probably be done dynamically. Since the batch request can contain metadata, instead of providing the `target_dtype` on initialization of the `AsType` node, one option would be to look at the request metadata and change the dtype to whatever has been requested.
Not sure if there would be a use for that. Might over complicate the problem.

Reply:
Maybe @funkey has some thoughts.
Otherwise `AsType` as is looks good

Reply:
I'd leave it as is. Automatically changing the `dtype` based on the request is a bit too much magic.